File: highPerformanceComputing/Project2-KNNClassifier/scratch/top100.py
# What is the goal
## The goal is to find the top 100 unique words in a text and store it in a csv file

# Step 1: Get rid of "frivolous" words
# Step 2: Find unique words and insert them into a dictionary { word : count }
# Step 3: Reduce the dictionary to the top 100

import os
import nltk
from collections import Counter

def print_tokens(tokens):
	for token in tokens:
		print(token)

def create_top_words(list):
	sorted_by_length = sorted(list, key = len)
	return sorted_by_length
#	return dic

if __name__ == "__main__":
	# open book and read it
	book = "hamlet"
	file_name = "./books/" + book + ".txt"
	file = open(file_name, "r")
	raw = file.read()

	# tokenize book
	tokens = nltk.word_tokenize(raw)
	
	top_100 = create_top_words(tokens)
	print(top_100)

#	for key, value in top_100.items():
#		print(key, value)

	file.close()

File: highPerformanceComputing/Project2-KNNClassifier/scratch/top100.c
// Author: Naoki Atkins

// What is the goal?
// 	The goal is to divide the file into several sections so that MPI
//	can divide the work amongst processes

//	Each process will find its own set of unique words and their frequencies

//	At the main thread, it will gather all of the results and find
// 	the most common set of unique words and the total of their frequencies

// 	This will be limited to 100 top words in a book

#include <stdio.h>
//#include <mpi.h>

int lineCount(char* fileName, FILE** readFile) {
	char fileChar[40], chr;
	int count = 0;
	chr = getc(*readFile);
	while (chr != EOF) {
		if (chr == "\n") count += 1;
	}
	return count; 
}

void readText(char* fileName, FILE** readFile) {
	if ((*readFile = fopen(fileName, "r")) == NULL ) return FILE_ERROR;
	else {
		
	}
}

int main(int argc, char** argv) {
	char bookName[] = "hamlet";
	char path[] = "./books/";
	char txt[] = ".txt";
	char interm[] = strcat(path, bookName);
	char filePath = strcat(filePath, txt);	

	printf("%s\n", filePath);

	return 0;
}
